plugin/src/App/Components/Tooltip.lua

```diff
@@ -180,31 +182,77 @@ function Trigger:willUnmount()
 	end
 end
 
+function Trigger:didUpdate(prevProps)
+	if prevProps.text ~= self.props.text then
+		-- Any existing popup is now invalid
+		self.props.context.removeTip(self.id)
+		self.showingPopup = false
+
+		-- Let the new text propagate
+		self:managePopup()
+	end
+end
+
+function Trigger:isHovering()
+	local rbx = self.ref.current
+	if rbx then
+		local pos = rbx.AbsolutePosition
+		local size = rbx.AbsoluteSize
+		local mousePos = self.mousePos
+
+		return
+			mousePos.X >= pos.X and mousePos.X <= pos.X + size.X
+			and mousePos.Y >= pos.Y and mousePos.Y <= pos.Y + size.Y
+	end
+	return false
+end
```

Comment:
This is simple to leave as a local function. The fewer methods we introduce now the easier it will be to migrate to hooks when the time comes.

Reply:
We'll cross that bridge when we get to it, if we ever do.

Reply:
This is a valid concern, but this feels reasonable for now and we'll have to handle a bunch of stuff like this when we migrate regardless. I think this is fine for now.